client/sites-dashboard/components/sites-dashboard.tsx

```diff
@@ -184,7 +184,7 @@ export function SitesDashboard( {
 
 	const { data: deletedSites = [] } = useSiteExcerptsQuery(
 		[],
-		( site ) => ! site.options?.is_domain_only,
+		( site ) => ! site.options?.is_domain_only && !! site?.is_deleted,
```

Comment:
I feel like this logic probably belongs inside `useSiteExcerptsQuery()`. The called has passed the `'deleted'` option, and so it feels like the query hook should hide all the messiness that comes from the fact that this data might actually be loaded from the redux cache.

The `is_domain_only` filtering would stay in the component, but the `is_deleted` check would be inside `useSiteExcerptsQuery`

Reply:
Another reason to do this logic in `useSiteExcerptsQuery` is because there are multiple places which make the request for deleted sites and they'd all need this same logic.

For example:

https://github.com/Automattic/wp-calypso/blob/867d91b103b19b7bd76bdd55686b94fdcb177c44/client/sites-dashboard-v2/index.tsx#L128